### ISSUE TITLE
Fix false negatives for ignore: ["inside-block"] in rule-empty-line-before

### ISSUE THREAD

**Issue by gucong3000** [stylelint/stylelint#2891]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Cannot monitor the missing blank line before rule in HTML.

> Which rule, if any, is this issue related to?

`rule-empty-line-before`

> What CSS is needed to reproduce this issue?

```html
<style>
.foo {
  color: pink;
}
.bar {
  color: pink;
}
</style>
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "rule-empty-line-before": ["always", {
      "ignore": "inside-block"
    }]
  }
}
```

> Which version of stylelint are you using?

8.1.1

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI
```shell
stylelint "**/*.html" --custom-syntax postcss-html
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

standard syntax in HTML.

> What did you expect to happen?

The following warnings were flagged in line that `.bar {`

> What actually happened (e.g. what warnings or errors you are getting)?

No warnings to be flagged.

**Comment by gucong3000** [comment]
@jeddy3
Maybe we need a new label. `non-standard Syntax: html`

**Comment by evilebottnawi** [comment]
@gucong3000 `non-standard syntax: html` is misleading, because `css` in `style` tag is standard syntax. Maybe `processor: html`

**Comment by jeddy3** [comment]
> Maybe we need a new label. non-standard syntax: html or some other description.

Done.

> because css in style tag is standard syntax. Maybe processor: html

Ummm. Good point. 

`postcss-html` is being used via the `--custom-syntax` option, right?:

```
stylelint "**/*.html" --custom-syntax postcss-html
```

What's the relationship/differences between `postcss-html` and the html processor?

**Comment by gucong3000** [comment]
> `postcss-html` is being used via the `--custom-syntax` option, right?:

yes.

> What's the relationship/differences between `postcss-html` and the html processor?

`postcss-html` can restore HTML files from AST. This means `postcss-html` compatible with that the feature of stylelint  - autofixing.

**Comment by jeddy3** [comment]
> postcss-html can restore HTML files from AST. This means postcss-html compatible with that the feature of stylelint - autofixing.

Thanks for explaining that. And am I right in thinking that not being able to do that is a current limitation of the processes? If so, is this a limitation that can be overcome?
